pandas/src/parser/tokenizer.c

```diff
@@ -679,7 +680,16 @@ static int parser_buffer_bytes(parser_t *self, size_t nbytes) {
     }
 
 int skip_this_line(parser_t *self, int64_t rownum) {
-    if (self->skipset != NULL) {
+    int should_skip;
+    PyGILState_STATE state;
+
+    if (self->skipfunc != NULL) {
+        state = PyGILState_Ensure();
+        should_skip = PyObject_IsTrue(PyObject_CallFunction(
```

Comment:
`PyObject_CallFunction` returns a new reference, so I think you need to capture the result and decref it (`Py_DECREF`).  Related to my comment on the tests, you may also need some error-trapping here in case the function fails.

https://docs.python.org/2/c-api/object.html#c.PyObject_CallFunction